initproducerid_test.go

```diff
@@ -7,40 +7,33 @@ import (
 	"testing"
 	"time"
 
-	"github.com/segmentio/kafka-go/protocol"
+	ktesting "github.com/segmentio/kafka-go/testing"
 )
 
 func TestClientInitProducerId(t *testing.T) {
-	client, shutdown := newLocalClient()
-
-	supported, err := isAPIKeySupported(context.Background(), client, protocol.InitProducerId)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if !supported {
-		t.Log("Skipping test for unsupported broker.")
+	if !ktesting.KafkaIsAtLeast("0.11.0") {
 		return
 	}
+	clientInitial, shutdownInitial := newLocalClient()
+	defer shutdownInitial()
```

Comment:
If you gave these different names from `client, shutdown := newClient(` to prevent collision in the `defer shutdown()` there's no need as it's actually safe to re-use the variable `shutdown` since `defer` is hanging on to the pointer of the first `shutdown` function

Reply:
Interesting. TIL.